### PR TITLE
Refactor gulp-core-build to allow for better task extensibility.

### PR DESCRIPTION
**Pull request by iclanton**

### CODE REVIEW

File: core-build/gulp-core-build-serve/src/TrustCertTask.ts [hunk 1]
```diff
@@ -13,7 +13,9 @@ import { ICertificate } from './certificates';
  *  and signed, but the user must trust it manually.
  */
 export default class TrustCertTask extends GulpTask<{}> {
-  public name: string = 'trust-cert';
```
Comment (nickpape-msft) [inline core-build/gulp-core-build-serve/src/TrustCertTask.ts]: >public name: string = 'trust-cert'; [](start = 2, length = 35)

We override this in a couple of places... maybe we should keep name as a public string

File: common/reviews/api/gulp-core-build-webpack.api.ts [hunk 2]
```diff
@@ -7,19 +7,16 @@ interface IWebpackTaskConfig {
...
   // (undocumented)
   public isEnabled(buildConfig: IBuildConfig): boolean;
   // (undocumented)
   public loadSchema(): Object;
   // (undocumented)
-  public name: string;
-  // (undocumented)
   public readonly resources: Object;
-  // (undocumented)
-  public taskConfig: IWebpackTaskConfig;
```
Comment (nickpape-msft) [inline common/reviews/api/gulp-core-build-webpack.api.ts]: this is technically a breaking change. we manually set taskConfig in a few places in gulpfiles.

File: common/reviews/api/gulp-core-build-webpack.api.ts [hunk 3]
```diff
@@ -7,19 +7,16 @@ interface IWebpackTaskConfig {
...
+class WebpackTask<TExtendedConfig = {}> extends GulpTask<IWebpackTaskConfig & TExtendedConfig> {
+  constructor(extendedName?: string, extendedConfig?: TExtendedConfig);
   // (undocumented)
   public executeTask(gulp: typeof Gulp, completeCallback: (error?: string) => void): void;
   // (undocumented)
   public isEnabled(buildConfig: IBuildConfig): boolean;
   // (undocumented)
   public loadSchema(): Object;
   // (undocumented)
-  public name: string;
```
Comment (nickpape-msft) [inline common/reviews/api/gulp-core-build-webpack.api.ts]: we also manually set this at times... why remove these properties? they should still be publicly accessible

File: core-build/gulp-core-build-serve/src/TrustCertTask.ts (hunk 1 above)
Comment (iclanton) [inline core-build/gulp-core-build-serve/src/TrustCertTask.ts]: It's still public, it's just only exposed on `GulpTask`.

File: common/reviews/api/gulp-core-build-webpack.api.ts (hunk 2 above)
Comment (iclanton) [inline common/reviews/api/gulp-core-build-webpack.api.ts]: It's still public, it's just only exposed on `GulpTask`.

File: common/reviews/api/gulp-core-build-webpack.api.ts (hunk 3 above)
Comment (iclanton) [inline common/reviews/api/gulp-core-build-webpack.api.ts]: It's still public, it's just only exposed on `GulpTask`.

**Comment by nickpape-msft** [comment]
:shipit:

**Pull request merged**